src/utils/calculateDuration.js

```diff
@@ -0,0 +1,9 @@
+import { FINISHED_STATES } from '@/utils/states'
+
+function isFinished(state) {
+  return FINISHED_STATES.includes(state)
+}
+
+export function duration(startTime, endTime, state) {
+  return endTime ? endTime : isFinished(state) ? startTime : null
```

Comment:
I would be in favor or more lines over nested ternaries. IMO This can be pretty hard to follow compared to 

```
if (endTime) {
   return endTime
}

if (isFinished(state)) {
    return startTime
}

return null
```

Reply:
+1